File: solutions/julia/scrabble-score/scrabble-score.jl
function letter_points(c)

    c = uppercase(c)

    if c ∈ ['A', 'E', 'I', 'O', 'U', 'L', 'N', 'R', 'S', 'T']
        return 1
    elseif c ∈ ['D', 'G']
        return 2
    elseif c ∈ ['B', 'C', 'M', 'P']
        return 3
    elseif c ∈ ['F', 'H', 'V', 'W', 'Y']
        return 4
    elseif c ∈ ['K']
        return 5
    elseif c ∈ ['J', 'X']
        return 8
    elseif c ∈ ['Q', 'Z']
        return 10
    else
        return 0
    end
end

function score(str)
    return sum(map(c -> letter_points(c), [c for c in str]))
end

File: solutions/cpp/scrabble-score/scrabble_score.cpp
#include "scrabble_score.h"

namespace scrabble_score {

    int letter_points(char c)
    {
        c = static_cast<char>(std::toupper(static_cast<unsigned char>(c)));

        switch (c) {
            case 'A': case 'E': case 'I': case 'O': case 'U':
            case 'L': case 'N': case 'R': case 'S': case 'T':
                return 1;

            case 'D': case 'G':
                return 2;

            case 'B': case 'C': case 'M': case 'P':
                return 3;

            case 'F': case 'H': case 'V': case 'W': case 'Y':
                return 4;

            case 'K':
                return 5;

            case 'J': case 'X':
                return 8;

            case 'Q': case 'Z':
                return 10;

            default:
                return 0;
        }
    }

    int score(const std::string& s)
    {
        int total = 0;
        for (char c : s)
            total += letter_points(c);
        return total;
    }

}  // namespace scrabble_score
